flowmachine/flowmachine/features/subscriber/topup_amount.py

```diff
@@ -0,0 +1,106 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+"""
+Class for calculating top-up statistics.
+"""
+
+import warnings
+
+from ..utilities.sets import EventsTablesUnion
+from .metaclasses import SubscriberFeature
+
+valid_stats = {"count", "sum", "avg", "max", "min", "median", "stddev", "variance"}
+
+
+class TopUpAmount(SubscriberFeature):
+    """
+    This class calculates statistics associated with top-up recharge amounts.
+
+    Parameters
+    ----------
+    start, stop : str
+         iso-format start and stop datetimes
+    statistic : {'count', 'sum', 'avg', 'max', 'min', 'median', 'mode', 'stddev', 'variance'}, default 'avg'
+        Defaults to sum, aggregation statistic over the durations.
+    hours : 2-tuple of floats, default 'all'
+        Restrict the analysis to only a certain set
+        of hours within each day.
+    subscriber_identifier : {'msisdn', 'imei'}, default 'msisdn'
+        Either msisdn, or imei, the column that identifies the subscriber.
+    subscriber_subset : str, list, flowmachine.core.Query, flowmachine.core.Table, default None
+        If provided, string or list of string which are msisdn or imeis to limit
+        results to; or, a query or table which has a column with a name matching
+        subscriber_identifier (typically, msisdn), to limit results to.
+    tables : str or list of strings, default 'events.topups'
+        Can be a string of a single table (with the schema)
+        or a list of these. The keyword all is to select all
+        subscriber tables
+
+    Examples
+    --------
+
+    >>> s = TopUpAmount("2016-01-01", "2016-01-08")
+    >>> s.get_dataframe()
+
+          subscriber       value
+    bjY7mMXxE3zMoelO    5.580000
+    dqEQWNwAYEr4Mk3e    5.150000
+    BNxWrJ9mqMJ83Mzk    3.968000
+    346j7Nq67nvXZR0m    6.265455
+    7XebRKr35JMJnq8A    3.748750
+                 ...         ...
+    """
+
+    def __init__(
+        self,
+        start,
+        stop,
+        statistic="avg",
+        *,
+        subscriber_identifier="msisdn",
+        hours="all",
+        subscriber_subset=None,
+        tables="events.topups",
```

Comment:
I think don't allow this to be configurable, and should it become necessary to be more flexible, add the flexibility.